src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/world/ModuleSurround.kt

```diff
@@ -0,0 +1,403 @@
+         */
+        @Suppress("unused")
+        private val tickHandler = handler<GameTickEvent>(priority = 10) {
+            if (!placer.crystalDestroyer.enabled || addExtraLayerBlocks) {
+                return@handler
+            }
+
+            broken.clear()
+            placer.blocks.filter { !it.value }.keys.forEach { pos ->
+                val breakingProgressions = mc.worldRenderer.blockBreakingProgressions[pos.asLong()] ?: return@forEach
+
+                val breakingInfo = breakingProgressions.lastOrNull { it.actorId != player.id } ?: return@forEach
+                val stage = breakingInfo.stage
+                if (stage < minDestroyProgress) {
+                    return@forEach
+                }
+
+                if (ExtraLayer.enabled && stage > 0) {
+                    broken.add(pos)
+                }
+
+                val blockedResult = pos.isBlockedByEntitiesReturnCrystal()
+                val crystal = blockedResult.value() ?: return@forEach
+
+                placer.crystalDestroyer.currentTarget = crystal
+                if (placer.crystalDestroyer.currentTarget == crystal) {
+                    return@handler
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Manually triggers the protection mechanism [Protect.ExtraLayer].
+     */
+    private val addExtraLayer by bind("AddExtraLayer", GLFW.GLFW_KEY_UNKNOWN)
+
+    init {
+        tree(Protect)
+    }
+
+    private val filter by enumChoice("Filter", Filter.WHITELIST)
+    private val blocks by blocks("Blocks", DEFAULT_BLOCKS)
+    private val placer = tree(BlockPlacer("Placing", this, Priority.NORMAL, { filter.getSlot(blocks) }))
+
+    private var addExtraLayerBlocks = false
+    private var startY = 0.0
+    private var centerPos: DoubleDoublePair? = null
+
+    init {
+        // for this module, support should by default be able to use obsidian
+        placer.support.blocks.addAll(DEFAULT_BLOCKS)
+    }
+
+    override fun enable() {
+        if (center) {
+            CommandCenter.center = true
+        }
+
+        startY = player.pos.y
+        val centerBlockPos = player.blockPos.toCenterPos()
+        centerPos = DoubleDoubleImmutablePair(centerBlockPos.x, centerBlockPos.z)
+    }
+
+    override fun disable() {
+        placer.disable()
+        addExtraLayerBlocks = false
+    }
+
+    @Suppress("unused")
+    val keyHandler = handler<KeyEvent> {
+        if (it.key != addExtraLayer.boundKey) {
+            return@handler
+        }
+
+        val action = addExtraLayer.action
+        if (it.action == GLFW.GLFW_PRESS) {
+            addExtraLayerBlocks = if (action == InputBind.BindAction.TOGGLE) {
+                !addExtraLayerBlocks
+            } else { // hold
+                true
+            }
+        } else if (it.action == GLFW.GLFW_RELEASE && action == InputBind.BindAction.HOLD) {
+            addExtraLayerBlocks = true
+        }
```

Comment:
This could be a part of the bind-value class. Like `newState = BindValue.getNewStateForEvent(oldState)`

Reply:
Thanks for pointing this out. This also made me realize that I was using the wrong event.